### PR TITLE
Bc/open link in private

### PR DESCRIPTION
**Pull request by ben-c-at-moz**
# Description

Create test to check that links can be opened in private windows through the context menu.

## Bugzilla bug ID

**Testrail:** https://testrail.stage.mozaws.net/index.php?/cases/view/101662
**Link:** https://bugzilla.mozilla.org/show_bug.cgi?id=1905177

## Type of change

Please delete options that are not relevant.

- [x] New Test
- [x] Other Changes -- Fix one of the ContextMenu methods

# How does this resolve / make progress on that bug?

Test complete

### CODE REVIEW

File: modules/browser_object_context_menu.py [hunk 1]
```diff
@@ -17,9 +19,20 @@ def get_context_item(self, item: str) -> WebElement:
...
+    ) -> BasePage:
         """
         Clicks the context item.
         """
         with self.driver.context(self.driver.CONTEXT_CHROME):
-            context_element.click()
+            self.fetch(reference, labels=labels).click()
+            return self
+
+    def click_and_hide(
```
Comment (sarinali) [inline modules/browser_object_context_menu.py:32]: much needed!

File: modules/page_base.py [hunk 2]
```diff
@@ -586,15 +586,15 @@ def hide_popup_by_class(self, class_name: str) -> None:
...
+        node = self.fetch(reference, labels=labels)
         script = """var element = arguments[0].parentNode;
                  if (element && element.hidePopup) {
                     element.hidePopup();
                  }"""
-        if chrome:
-            with self.driver.context(self.driver.CONTEXT_CHROME):
-                self.driver.execute_script(script, node)
-        else:
+        with self.driver.context(self.context_id):
```
Comment (sarinali) [inline modules/page_base.py:597]: Is there ever going to be a case where we have to dismiss a context menu in content context? I was just wondering bc of the change here

**Review by sarinali — APPROVED** [review]
LGTM, can you just explain that one method to me?

File: modules/page_base.py (hunk 2 above)
Comment (ben-c-at-moz) [reply modules/page_base.py:597]: I don't believe it's even possible to spawn a context menu in the content context.

**Pull request merged**